src/cargo/ops/lockfile.rs

```diff
@@ -67,7 +67,7 @@ pub fn write_pkg_lockfile(ws: &Workspace<'_>, resolve: &mut Resolve) -> CargoRes
     if resolve.version() < ResolveVersion::default() {
         resolve.set_version(ResolveVersion::default());
         out = serialize_resolve(resolve, orig.as_deref());
-    } else if resolve.version() > ResolveVersion::default()
```

Comment:
Should we have a `max_stable` constructor so we centralize this?

Reply:
https://github.com/rust-lang/cargo/pull/12860